hank-server/src/main/java/com/liveramp/hank/ring_group_conductor/RingGroupUpdateTransitionFunctionImpl.java

```diff
@@ -244,26 +267,63 @@ private Integer computeNumReplicasFullyServingRelevantData(Map<Domain, Map<Integ
 
     // Compute num replicas fully serving for given host, which is the minimum of the number of replicas
     // fully serving across all partitions assigned to it (for relevant domains)
-    Integer result = null;
+    LiveReplicaStatus worstStatus = LiveReplicaStatus.OVER_REPLICATED;
+
     for (HostDomain hostDomain : host.getAssignedDomains()) {
       Domain domain = hostDomain.getDomain();
       // Only consider relevant domains
       if (relevantDomains.contains(domain)) {
         Map<Integer, Set<Host>> partitionToNumFullyServing = domainToPartitionToHostsFullyServing.get(hostDomain.getDomain());
         if (partitionToNumFullyServing == null) {
-          return 0;
+          return LiveReplicaStatus.UNDER_REPLICATED;
         }
         for (HostDomainPartition partition : hostDomain.getPartitions()) {
-          int numFullyServing = 0;
           if (partitionToNumFullyServing.containsKey(partition.getPartitionNumber())) {
-            numFullyServing = partitionToNumFullyServing.get(partition.getPartitionNumber()).size();
-          }
-          if (result == null || numFullyServing < result) {
-            result = numFullyServing;
+
```

Comment:
How about just accumulating the statuses into an `EnumSet<LiveReplicaStatus>` and then `Collections.min` them at the end? I think that may be easier to follow.